File: 01-Arrays/20250723-beginner/ts/solution.ts
// TODO: Implement the solution
// This function should find and return the maximum element in the given array
// Return 0 for empty arrays
export function arrayMax(arr: number[]): number {
    // TODO: Replace this with your implementation
    if (arr.length === 0) {
        return 0
    }
    let max = 0
    for (let i = 0; i < arr.length; i++) {
        if (i === 0) {
            max = arr[i]

        } else {
            if (arr[i] > max) {
                max = arr[i]
            }
        }
    }
    return max
}

File: 01-Arrays/20250723-beginner/go/solution.go
package main

// TODO: Implement the solution
// This function should find and return the maximum element in the given array
// Return 0 for empty arrays
func ArrayMax(arr []int) int {
	// TODO: Implement this function
	if len(arr) == 0 {
		return 0
	}
    var max int
    for i,item:= range arr {
        if i ==0{
            max = item
        } else {
            if item > max {
                max = item
            }
        }
    }
	return max
}

